### PR TITLE
[new-downloader] storage integration tests

### PR DESCRIPTION
**Pull request by kshalnev**
Интеграционные тесты перенесены из map-downloader

### CODE REVIEW

File: testing/testingmain.cpp [hunk 1]
```diff
@@ -32,6 +32,33 @@
...
+  QAPP::exec();
+# endif
+#endif
+}
+
+void StopEventLoop()
+{
+#ifdef OMIM_OS_IPHONE_DEVICE
+  CFRunLoopStop(CFRunLoopGetMain());
+#else
```
Comment (ygorshenin) [inline testing/testingmain.cpp]:
```c++
#if defined(...)
...
#elif defined(...)
...
#endif
```

File: testing/testingmain.cpp [hunk 2]
```diff
@@ -32,6 +32,33 @@
   #endif
 #endif
 
+namespace testing
+{
+
+void RunEventLoop()
+{
+#ifdef OMIM_OS_IPHONE_DEVICE
```
Comment (ygorshenin) [inline testing/testingmain.cpp]:
```c++
#if defined(...)
...
#elif defined(...)
...
#endif
```

File: testing/testing.hpp [hunk 3]
```diff
@@ -27,6 +27,12 @@ namespace my
   }
 }
 
+namespace testing
+{
+  void RunEventLoop();
```
Comment (ygorshenin) [inline testing/testing.hpp]: Wrong indent - no spaces before these functions.

File: storage/storage_integration_tests/storage_update_tests.cpp [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+string GetCountriesTxtWebUrl(string const version)
+{
+  return kTestWebServer + "/direct/" + version + "/countries.txt";
+}
+
+string GetCountriesTxtFilePath()
+{
+  return GetPlatform().WritableDir() + "countries.txt";
```
Comment (ygorshenin) [inline storage/storage_integration_tests/storage_update_tests.cpp]: Please use my::JoinFoldersToPath().

File: storage/storage_integration_tests/storage_downloading_tests.cpp [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  }
+  catch (InterruptException const &)
+  {}
+
+  // Continue download
+
+  Storage storage(COUNTRIES_MIGRATE_FILE);
+
+  auto onProgressFn = [](TCountryId const & countryId, LocalAndRemoteSizeT const & mapSize) {};
+  auto onChangeCountryFn = [&](TCountryId const & countryId)
```
Comment (ygorshenin) [inline storage/storage_integration_tests/storage_downloading_tests.cpp]: This lambda is the same as the lambda above. Please, remove this code duplication.

File: storage/storage_integration_tests/storage_downloading_tests.cpp [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  {
+    if (!storage.IsDownloadInProgress())
+      testing::StopEventLoop();
+  };
+
+  storage.Init(Update);
```
Comment (ygorshenin) [inline storage/storage_integration_tests/storage_downloading_tests.cpp]: These 5 lines are almost the same as the storage initialization above. Please, get rid of this code duplication.

File: storage/storage_integration_tests/storage_downloading_tests.cpp (hunk 5 above)
Comment (kshalnev) [inline storage/storage_integration_tests/storage_downloading_tests.cpp]: Different storage instances are bound to the lambda, therefore these lambda must be different.

Comment (ygorshenin) [inline storage/storage_integration_tests/storage_downloading_tests.cpp]: Why? The first storage will be destroyed after the try block, so the second can re-use the same lambda safely.
```c++
auto onProgress = [](storage, countryId, mapSize)
{
  if (!storage.IsDownloadInProgress())
    testing::StopEventLoop();
};
...
try {
   storage.Subsribe(bind(onProgress, ref(storage), _1, _2));
   ...
}
...
storage.Subscribe(bind(onProgress, ref(storage), _1, _2));
...
```

Comment (kshalnev) [inline storage/storage_integration_tests/storage_downloading_tests.cpp]: yes, it is good.

File: testing/testingmain.cpp (hunk 1 above)
Comment (kshalnev) [inline testing/testingmain.cpp]: fixed

File: testing/testingmain.cpp (hunk 2 above)
Comment (kshalnev) [inline testing/testingmain.cpp]: fixed

File: testing/testing.hpp (hunk 3 above)
Comment (kshalnev) [inline testing/testing.hpp]: fixed

File: storage/storage_integration_tests/storage_update_tests.cpp (hunk 4 above)
Comment (kshalnev) [inline storage/storage_integration_tests/storage_update_tests.cpp]: fixed

**Pull request merged**